packages/react-composites/src/composites/CallComposite/adapter/AzureCommunicationCallAdapter.ts

```diff
@@ -381,11 +381,8 @@ export class AzureCommunicationCallAdapter implements CallAdapter {
   public async leaveCall(forEveryone?: boolean): Promise<void> {
     await this.handlers.onHangUp(forEveryone);
     this.unsubscribeCallEvents();
-    this.call = undefined;
     this.handlers = createDefaultCallingHandlers(this.callClient, this.callAgent, this.deviceManager, undefined);
-    this.context.setCurrentCallId(undefined);
-    // Resync state after callId is set
-    this.context.updateClientState(this.callClient.getState());
+    this.call = undefined;
```

Comment:
Should `this.call` also be reset on line 133 then.

This is creating a disconnect between `this.callId` and `this.call`. That might cause other bugs, or is difficult to understand / remember at the least.

Reply:
this.call is part of the call adapter, this.callId is part of the callContext